### ISSUE TITLE
CA1826 refactoring looses `!` operator

### ISSUE THREAD

**Issue by MaceWindu** [dotnet/roslyn-analyzers#5795]
### Analyzer

**Diagnostic ID**: [CA1826](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1826): `Use property instead of Linq Enumerable method`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 6.0.1 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 6.0.101](https://dotnet.microsoft.com/download/dotnet/6.0)

### Describe the bug

When collection had `!` operator applied, it is lost after refactoring leading to NRT errors.

### Steps To Reproduce

```cs
public class Test
{
    public static IReadOnlyList<string>? Strings { get; }

    public static string Method()
    {
        return Strings!.Last();
    }
}
```

### Expected behavior
```cs
public class Test
{
    public static IReadOnlyList<string>? Strings { get; }

    public static string Method()
    {
        return Strings![Strings.Count - 1];
    }
}
```

### Actual behavior

```cs
public class Test
{
    public static IReadOnlyList<string>? Strings { get; }

    public static string Method()
    {
        // CS8602
        return Strings[Strings.Count - 1];
    }
}
```
